### PR TITLE
OGRSQL: Add SELECT * EXCLUDE (...)

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Add `SELECT * EXCLUDE (...)` as described in #8668.

Still need to address some memory leaks. 

## What are related issues/pull requests?

#8668

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: autotest/ogr/ogr_sql_test.py [hunk 1]
```diff
@@ -1551,3 +1551,173 @@ def test_ogr_sql_min_max_string_field(data_ds):
...
+
+        assert sql_lyr.GetLayerDefn().GetGeomFieldCount() == 2
+
+
+def test_ogr_sql_select_except_named_geometry():
+
+    mem_ds = ogr.GetDriverByName("Memory").CreateDataSource("my_ds")
+    mem_lyr = mem_ds.CreateLayer("my_layer")
+
+    mem_lyr.GetLayerDefn().GetGeomFieldDefn(0).SetName("named_geom")
```
Comment (rouault) [inline autotest/ogr/ogr_sql_test.py:1633]: Changing the name (or other property) of a field or geometry field after it has been created is in the unspecified behaviour category
should rather be something along:
```suggestion
    mem_lyr = mem_ds.CreateLayer("my_layer", geom_type = ogr.wkbNone)
    mem_lyr.CreateGeomField(ogr.GeomFieldDefn("named_geom", ogr.wkbUnknown))
```

File: ogr/swq_select.cpp [hunk 2]
```diff
@@ -1221,7 +1240,7 @@ CPLErr swq_select::parse(swq_field_list *field_list,
         }
     }
 
-    /* -------------------------------------------------------------------- */
+    //    /* --------------------------------------------------rc_------------------ */
```
Comment (rouault) [inline ogr/swq_select.cpp:1243]:
```suggestion
    /* -------------------------------------------------------------------- */
```

**Review by rouault — COMMENTED** [review]
Very cool! We'd need a bit of documentation in https://gdal.org/user/ogr_sql_dialect.html (with a note that it is available *only* in OGR SQL currently, and not in the SQLite or most other native RBDMS native SQL)

**Comment by dbaston** [comment]
It looks like the tests are failing because of warnings in the generated `swq_parser.cpp`, e.g. https://github.com/OSGeo/gdal/actions/runs/6891452376/job/18746536270?pr=8675#step:8:2538 and https://github.com/OSGeo/gdal/actions/runs/6871181938/job/18687534919?pr=8675#step:5:15. I guess I should modify the generated file to avoid the warnings?

**Comment by rouault** [comment]
> I guess I should modify the generated file to avoid the warnings?

that or, suppress the MSVC C4505 warning in CMakeLists.txt for that file + ignore the cppcheck warning in scripts/cppcheck.sh

**Comment by rouault** [comment]
> that or, suppress the MSVC C4505 warning in CMakeLists.txt for that file + ignore the cppcheck warning in scripts/cppcheck.sh

@dbaston Just curious how you fixed those issues as I don't see any special hacks in CMakeLists.txt/cppchecks.h

**Comment by dbaston** [comment]
I edited the generated `swq_parser.cpp` to remove an always-true assert and a static function that was never called.

**Comment by rouault** [comment]
> I edited the generated `swq_parser.cpp` to remove an always-true assert and a static function that was never called.

ok, we'll have to redo that the next time we regenerate the file, but that might be in many years :-) so OK as a resolution

**Pull request merged**